pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -71,19 +72,41 @@ type Applier interface {
 	ApplyRuntimeObject(obj runtime.Object, scheme *runtime.Scheme) (hiveresource.ApplyResult, error)
 }
 
-// Add creates a new SyncSet controller and adds it to the manager with default RBAC. The manager will set fields on the
+// Add creates a new SyncSetInstance controller and adds it to the manager with default RBAC. The manager will set fields on the
 // controller and start it when the manager starts.
 func Add(mgr manager.Manager) error {
-	return AddToManager(mgr, NewReconciler(mgr))
+	logger := log.WithField("controller", controllerName)
+
+	// Get hiveConfig and parse hiveConfig.Spec.SyncSetReapplyInterval
+	tempClient, err := client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme()})
+	if err != nil {
+		return err
+	}
+	hiveConfig := &hivev1.HiveConfig{}
+	err = tempClient.Get(context.TODO(), types.NamespacedName{Name: hiveConfigName}, hiveConfig)
```

Comment:
The controllers should never get the HiveConfig. The hive-operator should pass the values from the HiveConfig into the controllers via environment variables or command-line arguments .

Reply:
Updated to set an environment variable from hiveconfig via the operator